core/src/stages/compute_ik.cpp

```diff
@@ -119,6 +120,16 @@ bool isTargetPoseColliding(const planning_scene::PlanningScenePtr& scene, Eigen:
 		}
 	}
 
+	// disable collision checking between collision objects and links not part of the rigid target group
+	const auto& world_object_ids = scene->getWorld()->getObjectIds();
+	for (const auto& robot_link : scene->getRobotModel()->getLinkModelsWithCollisionGeometry()) {
+		if (std::find(target_links.begin(), target_links.end(), robot_link) == target_links.end()) {
+			for (const auto& object_id : world_object_ids) {
+				acm.setEntry(object_id, robot_link->getName(), true);
+			}
+		}
+	}
```

Comment:
As far as I understand, this PR wants to consider collisions of end-effector links with collision objects only, for the pre-check in `isTargetPoseColliding()`. I agree that unrelated group links (e.g. another arm) should be ignored because they will be ignored later as well.
However, collisions of links rigidly connected to the root link of the current group should be still considered, shouldn't they? That's what the code in [lines 107-121](https://github.com/ros-planning/moveit_task_constructor/pull/117/files#diff-32b368f8353f812a564f161db5e3f48fR107-R121) was doing. With your PR this code actually would become superfluous.